### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by sander-adhese**

### CODE REVIEW

**Comment by sander-adhese** [comment]
Still to do (now or in later iterations)
- usersync & usersync tests
- gdpr
- reference
- write more tests

**Comment by sander-adhese** [comment]
If you want to test this:

1. Check out the project
2. Run go build (after installing everything for go)
3. Run ./prebid-server
4. In another tab, curl

`curl -v http://localhost:8000/openrtb2/auction -d '{ "id": "sander", "imp": [{ "id": "sander", "banner": { "format": [{"w": 300, "h": 250}]}, "ext": { "adhese": { "account": "demo", "location": "_adhese_prebid_demo_", "format": "leaderboard", "targets": [{ "key": "ci", "value": ["gent", "brussels"] }, { "key": "ag", "value": ["55"] }, { "key": "tl", "value": ["all"] } ] } } }], "site": { "id": "test", "publisher": {"id": "123"}}}'
`

**Comment by sander-adhese** [comment]
There is also a file file adhese_test.go which run the jsons in adapters/adhese/adhesetest/

**Comment by paulvv** [comment]
OK

File: openrtb_ext/imp_adhese.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	DeliveryMultiples         string                     `json:"deliveryMultiples"`
+	DeliveryGroupId           string                     `json:"deliveryGroupId"`
+	AdspaceStart              string                     `json:"adspaceStart"`
+	AdspaceEnd                string                     `json:"adspaceEnd"`
+	SwfSrc5th                 string                     `json:"swfSrc5th"`
+	SwfSrc6th                 string                     `json:"swfSrc6th"`
+	AdDuration5th             string                     `json:"adDuration5th"`
+	AdDuration6th             string                     `json:"adDuration6th"`
+	Width3rd                  string                     `json:"width3rd"`
+	Width4th                  string                     `json:"width4th"`
```
Comment (mefjush) [inline openrtb_ext/imp_adhese.go:102]: do we really need to expose that kind of fields? what for?

File: openrtb_ext/imp_adhese.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+	ViewableImpressionCounter string                     `json:"viewableImpressionCounter"`
+	AdditionalCreatives       []AdheseAdditionalCreative `json:"additionalCreatives"`
+	Origin                    string                     `json:"origin"`
+	OriginData                json.RawMessage            `json:"originData"`
+	Auctionable               string                     `json:"auctionable"`
+	AdditionalViewableTracker string                     `json:"additionalViewableTracker"`
+	Extension                 Prebid                     `json:"extension"`
+}
+
+type AdheseAdditionalCreative struct {
```
Comment (mefjush) [inline openrtb_ext/imp_adhese.go:122]: the same goes for this - how is it supposed to be used in prebid context?

**Pull request merged**